src/Shared/Model/NuGetMetadata.cs

```diff
@@ -0,0 +1,167 @@
+    public long? DownloadCount { get; private set; }
+    
+    [JsonProperty(PropertyName = JsonProperties.IconUrl)]
+    [JsonConverter(typeof(SafeUriConverter))]
+    public Uri IconUrl { get; private set; }
+    
+    [JsonIgnore]
+    public PackageIdentity Identity => new(PackageId, new NuGetVersion(Version));
+    
+    [JsonProperty(PropertyName = JsonProperties.Version)]
+    public string Version { get; private set; }
+
+    [JsonProperty(PropertyName = JsonProperties.LicenseUrl)]
+    [JsonConverter(typeof(SafeUriConverter))]
+    public Uri LicenseUrl { get; private set; }
+    
+    [JsonProperty(PropertyName = JsonProperties.ProjectUrl)]
+    [JsonConverter(typeof(SafeUriConverter))]
+    public Uri ProjectUrl { get; private set; }
+    
+    [JsonProperty(PropertyName = JsonProperties.ReadmeUrl)]
+    [JsonConverter(typeof(SafeUriConverter))]
+    public Uri ReadmeUrl { get; private set; }
+    
+    [JsonIgnore]
+    public Uri ReportAbuseUrl { get; }
+    
+    [JsonProperty(PropertyName = "packageDetailsUrl")]
+    [JsonConverter(typeof(SafeUriConverter))]
+    public Uri PackageDetailsUrl { get; private set; }
+    
+    [JsonProperty(PropertyName = JsonProperties.Published)]
+    public DateTimeOffset? Published { get; private set; }
+    
+    [JsonProperty(PropertyName = JsonProperties.Owners)]
+    [JsonConverter(typeof(MetadataFieldConverter))]
+    public string Owners { get; private set; }
+    
+    [JsonProperty(PropertyName = JsonProperties.PackageId)]
+    public string PackageId { get; private set; }
+    
+    [JsonProperty(PropertyName = JsonProperties.RequireLicenseAcceptance, DefaultValueHandling = DefaultValueHandling.Populate)]
+    [DefaultValue(false)]
+    [JsonConverter(typeof(SafeBoolConverter))]
+    public bool RequireLicenseAcceptance { get; private set; }
+    
+    [JsonProperty(PropertyName = JsonProperties.Summary)]
+    public string Summary { get; private set; }
+    
+    [JsonProperty(PropertyName = JsonProperties.Tags)]
+    [JsonConverter(typeof(MetadataFieldConverter))]
+    public string Tags { get; private set; }
+    
+    [JsonProperty(PropertyName = JsonProperties.Title)]
+    public string Title { get; private set; }
+    
+    [JsonProperty(PropertyName = JsonProperties.Listed)]
+    public bool IsListed { get; private set; }
+    
+    [JsonProperty(PropertyName = JsonProperties.PrefixReserved)]
+    public bool PrefixReserved { get; private set; }
+    
+    [JsonIgnore]
+    public LicenseMetadata LicenseMetadata { get; }
+    
+    [JsonProperty(PropertyName = JsonProperties.Vulnerabilities)]
+    public IEnumerable<PackageVulnerabilityMetadata> Vulnerabilities { get; private set; }
+    
+    [JsonProperty(PropertyName = JsonProperties.SubjectId)]
+    public Uri CatalogUri { get; private set; }
+
+    // Constructor for json deserialization
+    [JsonConstructor]
+#pragma warning disable CS8618
+    public NuGetMetadata()
+#pragma warning restore CS8618
+    {}
```

Comment:
This seems a bit odd - why bother defining a constructor with `[JsonConstructor]` when it accepts no arguments? Could we simply delete these lines?

Reply:
That is necessary for the way Newtonsoft.json constructs the class from json, I believe. I think I couldn't get it working otherwise. I will mess around with it tomorrow.

Reply:
I tried removing it and the unit tests from NuGetProjectManagerTests started failing. It has to do with the json parsing. I think I had seen this used elsewhere before but can't find it.